perun-cli/getOidcConfig

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use Perun::Agent;
+use Getopt::Long qw(:config no_ignore_case);
+
+
+my @fields = ("client_id", "oidc_device_code_uri", "oidc_token_endpoint_uri", "oidc_token_revoke_endpoint_uri", "acr_values", "scopes", "perun_api_endpoint", "enforce_mfa");
```

Comment:
one thing that comes to my mind is if we have all the fields defined like this, then additional property (let's say we add some new url) would be skipped. theoretically users should always update their CLI tools if perun version changes, but it is not always the case. would you say this is better than iterating all the fields that perun sends? what do you think, @xflord , @cuadradek ? also, we probably wouldn't have to extend the OidcConfig.pm and this class if new property is added.

Reply:
You're probably right. I did it this way mainly to keep the order of fields (which ultimately doesn't matter, I believe?) Is there a way to iterate all the fields sent in perl, or are you thinking of sending the list of fields from backend as well?

Reply:
in ConfigAgent it would go like
`return Perun::Common::callManagerMethod('getPerunOidcConfig', '', @_)`

and in the script something like
> foreach my $key (keys %$oidcConfig) {
> 	my $value = $oidcConfig->{$key};
> 	print $key . ": " . $value . "\n";
> }

but some smart handling of the json will probably be needed, because we should be able to use the string quotes on string and not use them for integer and boolean values (enforceMfa)...

Reply:
Used json encoding instead, which adds quotes to strings. The returned fields now aren't always in the same order, however that shouldn't matter.